devtools/onion.c

```diff
@@ -109,7 +109,7 @@ static void do_generate(int argc, char **argv,
 	u8 *serialized = serialize_onionpacket(ctx, res);
 	if (!serialized)
 		errx(1, "Error serializing message.");
-	printf("%s\n", tal_hex(ctx, serialized));
+	printf("%s", tal_hex(ctx, serialized));
```

Comment:
Hmm, this is a bit awkward if you run it on the cmdline though?

Can we instead fix it up on the input side?

Reply:
Good point; I've rebased to leave the trailing newline but instead trim whitespace with 6a078de . Is this better?